File: Algorithms/p299_Bulls_and_Cows/p299_Bulls_and_Cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        vector<int> secret_cnt(10,0);
        vector<int> guess_cnt(10,0);
        int a_cnt = 0;
        int b_cnt = 0;
        for(int i=0;i<secret.size();++i){
            if(secret[i]==guess[i]){
                ++a_cnt;
            }else{
                ++secret_cnt[secret[i]-'0'];
                ++guess_cnt[guess[i]-'0'];
            }
        }
        for(int i=0;i<10;++i){
            b_cnt += min(secret_cnt[i],guess_cnt[i]);
        }
        return to_string(a_cnt)+"A"+to_string(b_cnt)+"B";
    }
};

File: Algorithms/p299_Bulls_and_Cows/p299_Bulls_and_Cows.py
class Solution(object):
    def getHint(self, secret, guess):
        """
        :type secret: str
        :type guess: str
        :rtype: str
        """
        secret_cnt = [0]*10
        guess_cnt = [0]*10
        a_cnt = 0
        b_cnt = 0
        for i in range(len(secret)):
            if secret[i]==guess[i]:
                a_cnt += 1
            else:
                secret_cnt[int(secret[i])] += 1
                guess_cnt[int(guess[i])] += 1
        for i in range(10):
            b_cnt += min(secret_cnt[i],guess_cnt[i])
        return str(a_cnt)+'A'+str(b_cnt)+'B'
